### PR TITLE
feat(rest/server): support TLS protocol

### PR DESCRIPTION
**Pull request by itohatweb**

### CODE REVIEW

File: src/rest/types/server.ts [hunk 1]
```diff
@@ -11,6 +11,13 @@ export interface RestServerOptions {
   maxRetryCount: number;
   /** The api version you would like to use */
   apiVersion?: number;
+  /** The TLS keys you would like to use */
+  keys?: {
+    /** File path to your public key */
+    public: string;
+    /** File path to your private key */
+    private: string;
```
Comment (ayntee) [inline src/rest/types/server.ts:19]: Can we rename these options to reduce ambiguity? `cert` and `key`, for example

File: src/rest/types/server.ts [hunk 2]
```diff
@@ -11,6 +11,13 @@ export interface RestServerOptions {
   maxRetryCount: number;
   /** The api version you would like to use */
   apiVersion?: number;
+  /** The TLS keys you would like to use */
+  keys?: {
```
Comment (ayntee) [inline src/rest/types/server.ts:15]: `tlsFiles` :sweat_smile:

File: src/rest/types/server.ts [hunk 3]
```diff
@@ -11,6 +11,13 @@ export interface RestServerOptions {
...
+    /** File path to your public key */
```
Comment (ayntee) [inline src/rest/types/server.ts:16]:
```suggestion
    /** Path of the server certificate file */
```

File: src/rest/types/server.ts [hunk 4]
```diff
@@ -11,6 +11,13 @@ export interface RestServerOptions {
...
+    public: string;
+    /** File path to your private key */
```
Comment (ayntee) [inline src/rest/types/server.ts:18]:
```suggestion
    /** Path of the server public key file. */
```

**Review by ayntee — APPROVED** [review]

File: src/rest/types/server.ts [hunk 5]
```diff
@@ -11,6 +11,13 @@ export interface RestServerOptions {
   maxRetryCount: number;
   /** The api version you would like to use */
   apiVersion?: number;
+  /** The TLS keys you would like to use */
+  tlsFiles?: {
+    /** Path of the server certificate file */
+    cert: string;
+    /** Path of the server public key file. */
```
Comment (ayntee) [inline src/rest/types/server.ts:18]:
```suggestion
    /** Path of the server key file. */
```

**Review by ayntee — DISMISSED** [review]

File: src/rest/types/server.ts [hunk 6]
```diff
@@ -11,6 +11,13 @@ export interface RestServerOptions {
   maxRetryCount: number;
   /** The api version you would like to use */
   apiVersion?: number;
+  /** The TLS keys you would like to use */
+  keys?: {
+    /** Path of the server certificate file (public key) */
+    cert: string;
+    /** Path of the server's private key file. */
+    privateKey: string;
```
Comment (ayntee) [inline src/rest/types/server.ts:19]:
```suggestion
    key: string;
```

File: src/rest/server.ts [hunk 7]
```diff
@@ -1,12 +1,18 @@
...
 import { RestServerOptions } from "./types/mod.ts";
 
 /** Begins an http server that will handle incoming requests. */
 export async function startRESTServer(options: RestServerOptions) {
-  const server = serve({ port: options.port });
+  const server = options.keys
+    ? serveTLS({
+      port: options.port,
+      certFile: options.keys.cert,
+      keyFile: options.keys.privateKey,
```
Comment (ayntee) [inline src/rest/server.ts:13]:
```suggestion
      keyFile: options.keys.key,
```

**Review by Skillz4Killz — APPROVED** [review]

**Comment by ayntee** [comment]
Can you add a link to resources about TLS connections in JSDoc comment block?

**Pull request merged**